enhancements/cguConditionEnhancement.md

```diff
@@ -0,0 +1,452 @@
+
+```yaml
+[root@hv2 ~]# oc get co/openshift-apiserver -oyaml | yq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:38:09Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:39:58Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:43:16Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Available"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:38:08Z",
+    "message": "All is well",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Upgradeable"
+  }
+]
+```
+
+- **Network operator** has below condition:
+
+```yaml
+[root@hv2 ~]# oc get co/network -ojson | jq '.status.conditions'
+[
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "Degraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "False",
+    "type": "ManagementStateDegraded"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:35:06Z",
+    "status": "True",
+    "type": "Upgradeable"
+  },
+  {
+    "lastTransitionTime": "2022-06-07T15:35:25Z",
+    "status": "False",
+    "type": "Progressing"
+  },
+  {
+    "lastTransitionTime": "2022-03-21T14:37:57Z",
+    "status": "True",
+    "type": "Available"
+  }
+]
+```
+
+## Proposal
+
+This section proposes minimalistic changes without breaking the current design pattern. The below proposal/recommendation consists of more granular `status.conditions`  based on all the verification we already have in TALM.
+
+**Considering**: backup,precaching is true and enable for upgrade is true as well. 
+
+### Workflow Description
+
+1. Condition type related to cluster existing in the managedcluster. The type could reflect on the existence of the targeted cluster to be present in the managedcluster. 
+
+  ```yaml
+  Type:    "ClusterValidation/ClustersSelected",
+  Status:  True/False,
+  Reason:  "",
+  Message: “”
+  ```
+
+2. Condition type: **BackupDone**
+3. a) Spec validation for the `managed policies`. Currently it gets checked both during pre-caching and for upgrade. This check can be done before starting an upgrade, irrespective of pre-caching being true or false. b) Platform related verification. 
+Point a) and b) can be merged into PreUpgradeValidation where we check managed policies existence, conformity of platform image etc.
+
+ ```yaml
+  Type:    "PreUpgradeValidation",
+  Status:  True/False,
+  Reason:  "MissingManagedPolicy/InvalidPlatformImage/ValidationComplete",
+  Message: “”
+  ```
+
+4. Condition type: **PrecacheSpecValid** and **PrecachingDone** which is already in the current design. For **PrecacheSpecValid**, we may omit the highlighted Reason from the [current spec](#precachingSpec)
```

Comment:
Here, for PrecacheSpecValid type, we are only removing 2-3 Reason which is redundant for preUpgradeValidation. Since, the condition type is consistent and it will remain as it is, it seems to be backward compatible.